Add LWC integration back

animeavi opened the pull request:
The [Modern-LWC](https://github.com/Tsuser1/Modern-LWC) devs kindly added Factions support at my request recently so I decided to dig the old LWC integration code for Factions and make it work again.

I made some changes like clearing barrels as well as chests (made it a string comparison so it's still compatible with 1.13) and moving the config toggles to config.yml.

animeavi committed:
Add LWC integration back

animeavi committed:
Move LWC settings to config.yml

CmdrKittens commented:
I'm unfamiliar with LWC. Can LWC not be used on other item-holding blocks like trapped chests, furnaces, etc?

CmdrKittens commented on src/main/java/com/massivecraft/factions/zcore/persist/MemoryBoard.java:
```diff
@@ -158,7 +159,18 @@ public void unclaimAllInWorld(String factionId, World world) {
+                }
+            }
+        } else {
+            flocationIds.removeFaction(factionId);
```
Unless I missed something, I believe running this only in the `else` means that the factionToLandMap will still contain that faction's old places.

animeavi replied:
To be honest I don't know the answer to this. I mostly added the old code back without much regard.
If you could point me in the right direction I can try to fix this.

animeavi commented:
Yes, LWC can protect blocks other than chests.
Full list [here](https://github.com/Tsuser1/Modern-LWC/blob/master/src/config/core.yml#L165)

I could add them if needed.

animeavi committed:
Remove LWC locks and Factions separately

animeavi commented on src/main/java/com/massivecraft/factions/zcore/persist/MemoryBoard.java:
```diff
@@ -165,12 +165,11 @@ public void clean(String factionId) {
-            flocationIds.removeFaction(factionId);
         }
+
+        flocationIds.removeFaction(factionId);
```
@CmdrKittens I believe this is the correct way to do this. Thoughts?

CmdrKittens replied:
Yeah, this should work out. 👍

CmdrKittens commented on modules/Main/pom.xml:
```diff
@@ -428,6 +434,12 @@
             <version>1.3</version>
             <scope>compile</scope>
         </dependency>
+        <dependency>
```
Twice?

CmdrKittens commented on src/main/java/com/massivecraft/factions/integration/LWCFeatures.java:
```diff
@@ -0,0 +1,72 @@
+
+    public static void setup() {
+        Plugin test = Bukkit.getServer().getPluginManager().getPlugin("LWC");
+        if (test == null || !test.isEnabled()) return;
```
Maybe swap out the `== null` for `instanceof LWCPlugin` to do both a nullcheck *and* confirm it's not going to somehow result in a cast exception?

CmdrKittens commented:
> Yes, LWC can protect blocks other than chests.
> Full list [here](https://github.com/Tsuser1/Modern-LWC/blob/master/src/config/core.yml#L165)
> 
> I could add them if needed.

Is there API exposed to get the list of blocks LWC is protecting (since it can be user configured)?

animeavi committed:
Unprotect every block type in LWC's config

animeavi committed:
Replace LWC null check with instanceof check

animeavi commented:
Yes I changed the code to unprotect every block type included in LWC's config now.
I accidentally added LWC twice to pom.xml, yes, removed it in this commit too.

CmdrKittens commented on src/main/java/com/massivecraft/factions/integration/LWCFeatures.java:
```diff
@@ -0,0 +1,81 @@
+        }
+    }
+
+    private static boolean isProtectedBlock(Block block) {
```
Perhaps just call this instead, ensuring it uses the same check LWC uses : https://github.com/Tsuser1/Modern-LWC/blob/22d8fa7a343749d07f9edcc9d6df2dfbdccc19ff/src/main/java/com/griefcraft/lwc/LWC.java#L1373-L1389

CmdrKittens commented:
If you can test out if it all works with that LWC method I mentioned in the comment 15 minutes ago, I think we're good to merge.

animeavi committed:
Use LWC's own method to check for protectable blocks

Also use a more appropriate name for the LWC integration methods, we're not removing just chests anymore.

animeavi commented:
> If you can test out if it all works with that LWC method I mentioned in the comment 15 minutes ago, I think we're good to merge.

Good catch, dunno how I missed that.
Changed to it in the latest commit. I did test it and it works just as well, if not better.

CmdrKittens commented:
Awesome, thanks. I'll look over and pull tonight.

CmdrKittens commented:
Check out build 151, I slightly re-arranged the code to reduce duplication, and renamed the class to match other integrations.

animeavi commented:
That's absolutely fine, I was using the name of the old code that used to be in Factions anyway.
